collect_app/src/main/java/org/odk/collect/android/database/entities/DatabaseEntitiesRepository.kt

```diff
@@ -355,42 +388,53 @@ class DatabaseEntitiesRepository(context: Context, dbPath: String) : EntitiesRep
     }
 
     private fun quote(text: String) = "\"$text\""
-
-    companion object {
-        private const val DATABASE_VERSION = 2
-    }
 }
 
-private class EntitiesDatabaseMigrator :
-    DatabaseMigrator {
+class EntitiesDatabaseMigrator : DatabaseMigrator {
 
     override fun onCreate(db: SQLiteDatabase) {
-        db.execSQL(
-            """
-            CREATE TABLE IF NOT EXISTS ${ListsTable.TABLE_NAME} (
-                    $_ID integer PRIMARY KEY, 
-                    ${ListsTable.COLUMN_NAME} text NOT NULL,
-                    ${ListsTable.COLUMN_HASH} text
-            );
-            """.trimIndent()
-        )
+        createDbForVersion(db, DATABASE_VERSION)
     }
 
     override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int) {
-        dropAllTablesFromDB(db)
+        if (oldVersion == 1) {
+            throw IllegalStateException("Cannot upgrade from this beta version. Please reinstall Collect!")
+        } else if (oldVersion == 2) {
```

Comment:
This shouldn't use an `if/else` structure, because in the future we may implement additional upgrades. It should be possible to upgrade step by step, for example, from version 2 to 3, then from 3 to 4 just like it works in `forms.db` and `instances.db`.